fedora_messaging/_session.py

```diff
@@ -160,14 +157,17 @@ def publish(self, message, exchange=None):
                 self._connection.close()
             raise ConnectionException(reason=e)
 
-    def _connect_and_publish(self, exchange, routing_key, body, properties):
+    def _connect_and_publish(self, exchange, message):
         if not self._connection or not self._channel:
             self._connection = pika.BlockingConnection(self._parameters)
             self._channel = self._connection.channel()
             if self._confirms:
                 self._channel.confirm_delivery()
 
-        self._channel.publish(exchange, routing_key, body, properties)
+        body = json.dumps(message.body).encode('utf-8')
```

Comment:
If you wanted, you could add the body and routing key as properties (called 'encoded_body' and 'encoded_routing_key' or something) so you don't have to Just Know the body needs to be utf-8 encoded.

Maybe even just add it as a private property ('_encoded_body', etc) since I can't think of a reason for anything outside this library needing this.